### PR TITLE
Provide a builder UI for a Flux Filter Node

### LINKED ISSUE #3497: IFQL Builder: Filter builder

### PR DESCRIPTION
**Pull request by lukevmorris**
Closes #3497

filter

  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] swagger.json updated (if modified Go structs or API)
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/flux/components/FilterConditions.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      </>
+    )
+  }
+}
+
+interface FilterConditionNodeProps {
+  node: FilterNode
+}
+
+/* tslint:disable */
```
[the remark this answers is not in the document]
Comment (lukevmorris) [reply ui/src/flux/components/FilterConditions.tsx]: Done.

File: ui/src/flux/components/FilterTagList.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+      bodyID,
+      declarationID,
+      onChangeArg,
+      onGenerateScript,
+      func: {id: funcID, args},
+    } = this.props
+    const {value, key: argKey} = args[0]
+
+    if (!this.clauseIsParseable) {
+      return (
```
[the remark this answers is not in the document]
Comment (lukevmorris) [reply ui/src/flux/components/FilterTagList.tsx]: Done.

File: ui/src/flux/components/FilterTagList.tsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+          <div className="no-results">No tag keys found.</div>
+        </div>
+      </div>
+    )
+  }
+
+  private reduceNodesToClause(
+    nodes,
+    conditions: FilterTagCondition[]
+  ): [FilterClause, boolean] {
```
[the remark this answers is not in the document]
Comment (lukevmorris) [reply ui/src/flux/components/FilterTagList.tsx]: Done.

File: ui/src/flux/components/FuncArgs.tsx [hunk 4]
```diff
@@ -25,7 +27,7 @@ export default class FuncArgs extends PureComponent<Props> {
 
     return (
       <div className="func-node--tooltip">
-        <div className="func-args">{this.renderJoinOrArgs}</div>
+        <div className="func-args">{this.renderArguments}</div>
```
[the remark this answers is not in the document]
Comment (lukevmorris) [reply ui/src/flux/components/FuncArgs.tsx]: Done.

File: ui/src/flux/components/FilterPreview.tsx [hunk 5]
```diff
@@ -1,66 +1,62 @@
...
+  }
+
+  public render() {
+    return <FilterConditions nodes={this.state.nodes} />
+  }
+}
+
+FilterPreview.defaultProps = {
+  filterString: '',
+}
```
[the remark this answers is not in the document]
Comment (lukevmorris) [reply ui/src/flux/components/FilterPreview.tsx]: Done.

**Comment by lukevmorris** [comment]
@121watts check it

File: ui/test/flux/components/FilterTagList.test.tsx [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+          {key: 'key2', operator: '==', value: 'val4'},
+        ],
+      }
+      const filterString = builder.buildFilterString(clause)
+      expect(filterString).toEqual(
+        '(r) => (r.key1 != "val1" AND r.key1 != "val2") AND (r.key2 == "val3" OR r.key2 == "val4")'
+      )
+    })
+  })
+})
```
Comment (121watts) [inline ui/test/flux/components/FilterTagList.test.tsx]: +1 to infinity

**Pull request merged**